core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponentSpec.scala

```diff
@@ -1350,6 +1350,175 @@ class CompactEntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatch
+      AttributeName.withDefaultNS("trixyVal") -> AttributeString(targetType)
+    ))
+    val sourceEntity3 = Entity("sourceEntity3", sourceType, Map(
+      AttributeName.withDefaultNS("trixySingletonList") -> AttributeValueList(Seq(
+        AttributeString(targetType)
+      ))
+    ))
+
+    // Insert all entities
+    insertAndGetAll(Seq(targetEntity1, sourceEntity1, sourceEntity2, sourceEntity3))
+
+    // these references does not exist in the attributes but makes the underlying rename query think so
+    runAndWait(q.insertReferences(wsid, Set(
+      RefMapping(sourceEntity1.toPointer, Set(targetEntity1.toPointer)),
+      RefMapping(sourceEntity2.toPointer, Set(targetEntity1.toPointer)),
+      RefMapping(sourceEntity3.toPointer, Set(targetEntity1.toPointer))
+    )))
+
+    // Execute renameEntityType and verify the result
+    val newTargetType = "newTargetType"
+    runAndWait(q.renameEntityType(wsid, targetType, newTargetType)) shouldBe 1
+
+    val newTargetEntity1 = targetEntity1.copy(entityType = newTargetType)
+    
+    // Verify entity types were updated in the ENTITY table
+    val newTargets = runAndWait(q.listEntities(wsid, newTargetType))
+    newTargets.map(_.toEntity) should contain theSameElementsAs Seq(newTargetEntity1)
+    
+    // Verify source is unchanged
+    val sourceEntities = runAndWait(q.listEntities(wsid, sourceType))
+    sourceEntities.map(_.toEntity) should contain theSameElementsAs Seq(
+      sourceEntity1,
+      sourceEntity2,
+      sourceEntity3
+    )
+  }
+  
+  it should "handle no references correctly" in withMinimalTestDatabase { _ =>
+    // Create target entities
+    val targetType = "targetType"
+    val targetEntity1 = Entity("targetEntity1", targetType, Map())
+
+    // Create source entities
+    val sourceType = "sourceType"
+    val sourceEntity1 = Entity("sourceEntity1", sourceType, Map(
+      AttributeName.withDefaultNS("ref") -> targetEntity1.toReference
+    ))
+
+    // Insert all entities
+    insertAndGetAll(Seq(targetEntity1, sourceEntity1))
+
+    runAndWait(q.insertReferences(wsid, Set(
+      RefMapping(sourceEntity1.toPointer, Set(targetEntity1.toPointer))
+    )))
+
+    // Execute renameEntityType, this time on sourceType which has no references
+    val newSourceType = "newSourceType"
+    val result = runAndWait(q.renameEntityType(wsid, sourceType, newSourceType))
+    
+    // Verify the entity was updated
+    result shouldBe 1
+    
+    // Get the updated entity
+    val updatedSource = runAndWait(q.getEntity(wsid, newSourceType, sourceEntity1.name))
+    
+    // Verify the entity exists with the new type
+    val newSourceEntity = sourceEntity1.copy(entityType = newSourceType)
+    updatedSource.map(_.toEntity) shouldBe Some(newSourceEntity)
+
+    // Verify the references were updated
+    runAndWait(q.getReferencesFrom(wsid, newSourceEntity.toPointer)) should contain theSameElementsAs Seq(targetEntity1.toPointer)
+  }
+
```

Comment:
These tests are comprehensive - no changes requested, just thinking out loud: you could also check the entity type metadata response to ensure that it contains the newType and expected counts and does not contain the oldType